tools/monitoring/mysql/operations.go

```diff
@@ -67,3 +67,37 @@ func rollbackTx(tx *sql.Tx, err error) error {
 	}
 	return fmt.Errorf("Statement execution failed: %v; rollback failed: %v", err, rollbackErr)
 }
+
+// CheckAlertCondition checks whether the given error pattern meets
+// the alert condition specified in config
+func CheckAlertCondition(errorPattern string, config *config.SelectedConfig, db *sql.DB) (bool, error) {
+	// the timestamp we want to start collecting logs
+	startTime := time.Now().Add(time.Minute * time.Duration(config.Period))
+
+	_, err := db.Query(`
+		CREATE VIEW Matched AS
+  	SELECT Jobname, PrNumber 
+  	FROM ErrorLogs
+  	WHERE ErrorPattern=? and TimeStamp > ?`,
```

Comment:
nit: Is the indent of SELECT and CREATE not aligned due to the formatter? Or does github web rendering spaces differently?  Could you align it similar as follows? Maybe a mix of tabs and spaces in the preceding whitespaces? 
```
	_, err := db.Query(`
                 CREATE VIEW Matched 
                 AS
                        SELECT Jobname, PrNumber 
                        FROM ErrorLogs
                        WHERE ErrorPattern=? and TimeStamp > ?`,
```

Reply:
do we need to create a view?